File: Semester3/Advanced Methods of Programming/Toy Language Interpreter C#/Assignement7/Assignement7/Model/CompoundStatement.cs
using Assignement7.Collection;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Assignement7.Model
{
    public class CompoundStatement : IStatement
    {
        private IStatement first;
        private IStatement second;

        public CompoundStatement(IStatement first, IStatement second)
        {
            this.first = first;
            this.second = second;
        }


        public ProgramState Execute(ProgramState state)
        {
            MyIStack<IStatement> stack = state.getExecutionStack();
            stack.Push(second);
            stack.Push(first);
            return null;
        }

        public override string ToString()
        {
            return first.ToString() + "; " + second.ToString();
        }
    }
}

File: Semester3/Advanced Methods of Programming/Toy Language Interpreter JavaFx/Assignment8/src/Model/Statement/CompoundStatement.java
package Model.Statement;

import Collection.Stack.MyIStack;
import Model.Exceptions.ToyLanguageInterpreterException;
import Model.ProgramState;

public class CompoundStatement implements IStatement {
    private IStatement first;
    private IStatement second;

    public CompoundStatement(IStatement first, IStatement second){
        this.first = first;
        this.second = second;
    }

    @Override
    public ProgramState execute(ProgramState state) throws ToyLanguageInterpreterException {
        MyIStack<IStatement> stack = state.getExecutionStack();
        stack.push(second);
        stack.push(first);
        return null;
    }

    @Override
    public String toString(){
        return first.toString() + "; " + second.toString();
    }
}
